src/hooks/useTxHistory.ts

```diff
@@ -1,8 +1,16 @@
-import { getTransactionHistory, TransactionListPage } from '@gnosis.pm/safe-react-gateway-sdk'
+import {
+  getTransactionHistory,
+  type SafeIncomingTransfersResponse,
+  type SafeModuleTransactionsResponse,
+  type SafeMultisigTransactionsResponse,
+  type TransactionListPage,
+} from '@gnosis.pm/safe-react-gateway-sdk'
+import { useRouter } from 'next/router'
 import { useAppSelector } from '@/store'
 import useAsync from './useAsync'
 import { selectTxHistory } from '@/store/txHistorySlice'
 import useSafeInfo from './useSafeInfo'
+import { getFilteredTxHistory, hasTxFilterQuery } from '@/components/transactions/TxFilterForm/utils'
```

Comment:
It's a bit weird to import these utils from a form component folder, don't you find? I would make a service for all things tx filter, and import them from there in this hook and in the component alike.